python/dgl/nn/pytorch/explain/pgexplainer.py

```diff
@@ -216,17 +220,71 @@ def train_step(self, graph, feat, tmp, **kwargs):
         Tensor
             A scalar tensor representing the loss.
         """
+        assert (
+            self.graph_explanation
+        ), '"explain_graph" must be True in initializing the module.'
+
         self.model = self.model.to(graph.device)
         self.elayers = self.elayers.to(graph.device)
 
-        pred = self.model(graph, feat, embed=False, **kwargs).argmax(-1).data
+        pred = self.model(graph, feat, embed=False, **kwargs)
+        pred = pred.argmax(-1).data
 
         prob, _ = self.explain_graph(
             graph, feat, tmp=tmp, training=True, **kwargs
         )
 
-        loss_tmp = self.loss(prob, pred)
-        return loss_tmp
+        loss = self.loss(prob, pred)
+        return loss
+
+    def train_step_node(self, nodes, graph, feat, tmp, **kwargs):
+        r"""Compute the loss of the explanation network
+
+        Parameters
+        ----------
+        nodes : int, iterable[int], tensor
+            The nodes from the graph used to train explanations, which cannot
+            have any duplicate value.
+        graph : DGLGraph
+            Input batched homogeneous graph.
```

Comment:
```suggestion
            Input homogeneous graph.
```

For node classification, the graph is typically not batched.

Reply:
gotcha, thanks